wearipedia/devices/underarmour/myfitnesspal_fetch.py

```diff
@@ -0,0 +1,248 @@
+import pandas as pd
+
+# Functions to generate data
+# These functions are used to generate data that closely mimic the
+# data that is generated by the MyFitnessPal API
+
+
+def goal_generator(self, days):
+    # creating an empty list to store the goals
+    goals = []
+
+    # for each day in the date range, we get the goals for that day
+    for day in days:
+        # using the client, we get the goals for the day
+        res = self.client.get_date(int(day.year), int(day.month), int(day.day)).goals
+        # we add the date to the goals
+        res["date"] = day
+        # we append the goals to the list
+        goals.append(res)
+    # we return the list of goals
+    return goals
+
+
+def daily_summary_generator(self, days):
+    # creating an empty list to store the daily summary
+    summary = []
+
+    # for each day in the date range, we get the daily summary for that day
+    for day in days:
+        # using the client, we get the daily summary for the day
+        res = self.client.get_date(int(day.year), int(day.month), int(day.day)).totals
+        # we add the date to the daily summary
+        res["date"] = day
+        # we append the daily summary to the list
+        summary.append(res)
+
+    # we return the list of daily summary
+    return summary
+
+
+def cardio_generator(self, days):
+    # creating an empty list to store the cardio exercises
+    cardio = []
+
+    # for each day in the date range, we get the cardio exercises for that day
+    for day in days:
+        # using the client, we get the cardio exercises for the day
+        res = (
+            self.client.get_date(int(day.year), int(day.month), int(day.day))
+            .exercises[0]
+            .get_as_list()
+        )
+        # we add the date to the cardio exercises
+        res = [{"day": day}] + res
+        # we append the cardio exercises to the list
+        cardio.append(res)
+    # we return the list of cardio exercises
+    return cardio
+
+
+def strength_generator(self, days):
+    # creating an empty list to store the strength exercises
+    strength = []
+
+    # for each day in the date range, we get the strength exercises for that day
+    for day in days:
+        # using the client, we get the strength exercises for the day
+        res = (
+            self.client.get_date(int(day.year), int(day.month), int(day.day))
+            .exercises[1]
+            .get_as_list()
+        )
+        # we add the date to the strength exercises
+        res = [{"day": day}] + res
+        # we append the strength exercises to the list
+        strength.append(res)
+    # we return the list of strength exercises
+    return strength
+
+
+def breakfast_generator(self, days):
```

Comment:
From 81-205, these meal generating functions are basically the same save for the different index into meals. I would recommend squeezing all three functions into just one generalized function that takes as a parameter the type of meal/index. 

Same should be done for the other generating functions.